src/rt-app_taskgroups.c

```diff
@@ -0,0 +1,365 @@
+#include <stdlib.h>
+#include <string.h>
+#include <mntent.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <stdio.h>
+#include <errno.h>
+
+#include "config.h"
+#include "rt-app_utils.h"
+#include "rt-app_taskgroups.h"
+
+typedef struct _taskgroup_ctrl_t
+{
+	char mount_point[FILENAME_MAX];
+} taskgroup_ctrl_t;
+
+static taskgroup_ctrl_t tg_ctrl;
+
+static int cgroup_check_cpu_controller(void)
+{
+	int dummy[3], ret = 0;
+	FILE *cgroups;
+	char buf[512];
+
+	cgroups = fopen("/proc/cgroups", "re");
+
+	if (!cgroups) {
+		perror("fopen");
+		goto err;
+	}
+
+	/* Ignore the first line as it contains the header. */
+	if (!fgets(buf, sizeof(buf), cgroups)) {
+		perror("fgets");
+		goto err;
+	}
+
+	while (!feof(cgroups)) {
+		if (fscanf(cgroups, "%s %d %d %d", buf, &dummy[0], &dummy[1],
```

Comment:
Looks like this question is still valid and unanswered. :-)

Reply:
According to the cgroups man page, this can be 0 - 
"4. This field contains the value 1 if this controller is enabled, or 0 if it has been disabled (via the cgroup_disable kernel command-line boot parameter)."

Reply:
That's why I came up with this fix. I already mention this command-line parameter further above.

Reply:
This conversation thread looks weird because I replied whilst dietmar was replying - he published after I wrote mine - but Github shows them ordered by writing time rather than publication time.